Services/Mail/test/ilMailOptionsTest.php

```diff
@@ -60,5 +67,90 @@ public function testConstructor() : void
         $this->assertSame($object->incoming_type, $mailOptions->getIncomingType());
         $this->assertSame($object->linebreak, $mailOptions->getLinebreak());
         $this->assertSame($object->cronjob_notification, $mailOptions->isCronJobNotificationEnabled());
+        $this->assertSame($object->absence_status, $mailOptions->getAbsenceStatus());
+        $this->assertSame($object->absent_from, $mailOptions->getAbsentFrom());
+        $this->assertSame($object->absent_until, $mailOptions->getAbsentUntil());
+        $this->assertSame($object->absence_ar_subject, $mailOptions->getAbsenceAutoResponderSubject());
+        $this->assertSame($object->absence_ar_body, $mailOptions->getAbsenceAutoResponderBody());
+    }
+
+    /**
+     * @dataProvider provideMailOptionsData
+     */
+    public function testIsAbsent(bool $absence_status, int $absent_from, int $absent_until, bool $result) : void
+    {
+        $userId = 1;
+        $object = new stdClass();
+        $object->cronjob_notification = false;
+        $object->signature = 'smth';
+        $object->linebreak = 0;
+        $object->incoming_type = 1;
+        $object->mail_address_option = 0;
+        $object->email = 'test@test.com';
+        $object->second_email = 'ilias@ilias.com';
+        $object->absence_status = $absence_status;
+        $object->absent_from = $absent_from;
+        $object->absent_until = $absent_until;
+        $object->absence_ar_subject = 'subject';
+        $object->absence_ar_body = 'body';
+        $database = $this->getMockBuilder(ilDBInterface::class)->getMock();
+        $queryMock = $this->getMockBuilder(ilDBStatement::class)->getMock();
+
+        $clockService = $this->getMockBuilder(ClockInterface::class)->disableOriginalConstructor()->getMock();
```

Comment:
```suggestion
        $clockService = $this->getMock(ClockInterface::class);
```

Bei Interfaces ist die Mock-Erstellung wie vorgeschlagen ausreichend.